copy/copy.go

```diff
@@ -546,6 +551,25 @@ func (c *copier) copyOneImage(ctx context.Context, policyContext *signature.Poli
 		return nil, "", "", errors.Wrapf(err, "Error initializing image from source %s", transports.ImageName(c.rawSource.Reference()))
 	}
 
+	options.DestinationManifestCheck = true // TODO: Forced enablement for initial PR testing
+	if options.DestinationManifestCheck && options.SignBy == "" {
```

Comment:
Should verify that `options.DestinationManifestCheck` is not set, if `options.SignBy` is set? Since both of these options are exclusive (with the current implementation).

The alternative would be to leave this decision to the client (which might choose to only set `options.DestinationManifestCheck` if `options.SignBy` is not set). And alway skip the copy exclusively based upon `options.DestinationManifestCheck`.

I don't think `options.DestinationManifestCheck && options.SignBy == "" ` is the right approach, as it allows the client to request skipping the manifest copy when that is not possible.

Reply:
The way I think about it, the caller is not making a _request_ that the code here must honor — the caller does not want the `copy.Image` call to _fail_ just because a full image must be copied. The caller is only is providing `AssumeImageAlreadyExistsAtDestination` information that this code could completely ignore, or use in any way it wants to optimize the behavior. 

From that point of view, it is perfectly fine for the client to provide “image probably already exists” information, together with a request for an operation (format conversion, encryption, …) which we don’t currently support in the optimization. Maybe such an optimization can be implemented in the future; maybe not. The caller can provide that information just the same, and if some future optimization arrives, the caller will benefit at that time.

Reply:
> The way I think about it, the caller is not making a request that the code here must honor

Thanks, nice. I *was* thinking about it like this. But I agree, this is more of an 'optimization hint'.